src/tools/dev/scripts/bv_support/bv_vtk.sh

```diff
@@ -1052,32 +1052,47 @@ function apply_vtk9_vtkRectilinearGridReader_patch
   # patch vtkRectilinearGridReader.cxx, per this issue:
   # https://gitlab.kitware.com/vtk/vtk/-/issues/18447
    patch -p0 << \EOF
-*** IO/Legacy/vtkRectilinearGridReader.cxx.orig	Thu Jan 27 10:55:12 2022
---- IO/Legacy/vtkRectilinearGridReader.cxx	Thu Jan 27 11:01:04 2022
-***************
-*** 95,101 ****
-          break;
-        }
-  
-!       if (!strncmp(this->LowerCase(line), "dimensions", 10) && !dimsRead)
-        {
-          int dim[3];
-          if (!(this->Read(dim) && this->Read(dim + 1) && this->Read(dim + 2)))
---- 95,108 ----
-          break;
-        }
-  
-!       // Have to read field data because it may be binary.
-!       if (!strncmp(this->LowerCase(line), "field", 5))
-!       {
-!         vtkFieldData* fd = this->ReadFieldData();
-!         fd->Delete();
-!       }
-! 
-!       else if (!strncmp(this->LowerCase(line), "dimensions", 10) && !dimsRead)
-        {
-          int dim[3];
-          if (!(this->Read(dim) && this->Read(dim + 1) && this->Read(dim + 2)))
+--- IO/Legacy/vtkRectilinearGridReader.cxx.orig	2024-06-05 14:21:59.105807000 -0700
++++ IO/Legacy/vtkRectilinearGridReader.cxx	2024-06-05 14:26:30.561802000 -0700
+@@ -95,7 +95,16 @@
+         break;
+       }
+ 
+-      if (!strncmp(this->LowerCase(line), "dimensions", 10) && !dimsRead)
++      // If data file is binary and there FieldData is present, it
++      // must be read here, otherwise a ReadString will fail and the
++      // loop will terminate before reading dimensions.
++      if (!strncmp(this->LowerCase(line), "field", 5))
```

Comment:
If `"FieldData"` is the thing to be looking for, why compare to just `"field"` here? Also, if the case in `"FieldData"` is relevant, why compare with `LowerCase(line)`?

Reply:
Makes sense, I had originally copied the line from logic elsewhere in the file.
If I change it to 'fielddata', then I can probably remove the extra logic to terminate early. (also making it possibly easier to get Kitware to accept the  patch).
Will try that out. Thanks!

I have been confused by the use of LowerCase (or lack thereof in some cases).
I think I will keep with the LowerCase just for safety and more consistency with usage elsewhere in the file.

Reply:
> I have been confused by the use of LowerCase (or lack thereof in some cases). I think I will keep with the LowerCase just for safety and more consistency with usage elsewhere in the file.

👍🏻

Reply:
So, it does occur to me though...what if a plugin or data producer creates a folder in a variable hierarchy with the name "FieldData" or "fielddata"?

Reply:
As relates to this patch, I could keep the early-loop termination. I'm not sure why the logic exists to read every line of the file in this method. It's not needed, they are only trying to get dimensions and possibly extents.
But I don't know if the other methods where it is necessary to read the entire file, if 'FieldData' as part of the name for SCALARS, etc would parse correctly. I can add it to my tests to see what happens.  If we cannot avoid name collisions from data producers with VTK dataset keywords, we may have to add logic to readers of said producers to mitigate the problem.

Reply:
The patch must remain as it is written.

Here is the beginning of the binary file example for which the original patch was created.
`FIELD` is the main keyword. `FieldData` is secondary.

```
vtk output
BINARY
DATASET RECTILINEAR_GRID
FIELD FieldData 1
some_field_data 1 4 int
```

Reply:
Ok, thanks for looking into it and jumping on a fix so quickly. This is great!